sdk/core/azure-core/src/main/java/com/azure/core/util/HttpClientOptions.java

```diff
@@ -191,13 +212,76 @@ public HttpClientOptions setReadTimeout(Duration readTimeout) {
 
     /**
      * Gets the read timeout duration used when reading the server response.
+     * <p>
+     * By default the read timeout is 60 seconds.
      *
      * @return The read timeout duration.
      */
     public Duration getReadTimeout() {
         return getTimeout(readTimeout);
     }
 
+    /**
+     * Sets the maximum connection pool size used by the underlying HTTP client.
+     * <p>
+     * By default the maximum connection pool size is determined by the underlying HTTP client.
+     *
+     * @param maximumConnectionPoolSize The maximum connection pool size.
+     * @return The updated HttpClientOptions object.
+     * @throws IllegalArgumentException If {@code maximumConnectionPoolSize} is less than {@code 1}.
+     */
+    public HttpClientOptions setMaximumConnectionPoolSize(int maximumConnectionPoolSize) {
```

Comment:
To allow for unsetting the pool size we should use `Integer` and allow null values. If not null, we should check that it's greater than 0.